src/factory.ts

```diff
@@ -46,8 +47,17 @@ function isGitHubRelease(
   return command === 'github-release' && typeof options === 'object';
 }
 
-function run(runnable: ReleasePR | GitHubRelease) {
-  return runnable.run();
+function run(
+  runnable: ReleasePR,
+  cmd: 'run' | 'latestTag'
+): Promise<number | GitHubTag | undefined>;
+function run(
+  runnable: GitHubRelease,
+  cmd: 'run'
+): Promise<ReleaseResponse | undefined>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function run(runnable: any, cmd: any = 'run') {
```

Comment:
I wish we could avoid this `any` type, what if we just introduce a Runnable interface, that's something like:

```
interface Runnable {
  run: (): ReleaseResponse | undefined | number
}
```

☝️ which GitHubRelease and GitHubRelease implement? Then your mocks could just implement this.

Reply:
I'll spend some time thinking about this. I kind of reached for function overloads as a quick fix to support GitHubRelease.run ReleasePR.run, and now ReleasePR.latestTag inside of this `run` method. Reading the doc examples they use `any` for the actual method implementation. But I can avoid it with a big fat pile of unions and a type assertion in the implementation:

```suggestion
function run(
  runnable: ReleasePR | GitHubRelease,
  cmd: 'run' | 'latestTag'
): Promise<number | GitHubTag | ReleaseResponse | undefined> {
  if (runnable instanceof GitHubRelease) {
    return runnable[cmd as 'run']();  // type assertion less bad than `any`
  }
  return runnable[cmd]();
}
```

It still has the benefit of the type overloads to help enforce correct calling but also has a higher level of type correctness itself, just looks kinda ugly...

But maybe there's a better pattern for accomplishing this single method to dispatch the execution? I'll keep pondering :-)

Reply:
ended up refactoring this a bit more. renamed `run` to `call` since that's what this dispatcher is really doing now: calling a method on an instance.